packages/x-components/src/x-modules/facets/entities/filter-entity.factory.ts

```diff
@@ -36,7 +36,13 @@ export class FilterEntityFactory {
   /**
    * The registered modifiers grouped by the facetId to be applied.
    */
-  protected modifiers: Record<Facet['id'], FilterEntityModifier[]> = {};
+  protected modifiersByFacetId: Record<Facet['id'], FilterEntityModifier[]> = {};
+
+  /**
+   * The registered modifiers grouped by the filter `ModelName` to be applied.
+   */
+  protected modifiersByFilterModelName: Partial<Record<FilterModelName, FilterEntityModifier[]>> =
+    {};
```

Comment:
Do we need two different records to store the same thing? I don't think keys will overlap, so we can have just one list:

```typescript
protected modifiers: Record<Facet['id'] | FilterModelName, FilterEntityModifier[]> = {};
```

Then, we can get rid of the new register method, and just use the previous one:

```typescript
/**
   * Registers a list of modifiers to be used with the {@link FilterEntity | FilterEntities} of a
   * particular facet.
   *
   * @param modifierKey - The facet id or filter model name whose Entities will be modified.
   * @param modifiers - The list of modifiers to be registered.
   */
  registerModifier(
    modifierKey: Facet['id'] | FilterModelName,
    ...modifiers: FilterEntityModifier[]
  ): void {
    this.updateModifiers(this.modifiers, modifierKey, modifiers);
  }
```

And in `creatingFilterEntity()`

```typescript
const modifiers = isFacetFilter(filter)
  ? this.modifiers[filter.facetId] ?? this.modifiers[filter.modelName] ?? []
  : [];
```

Reply:
I was thinking the same when I saw the PR. It might be slightly easier for debuging issues if we have two separate records, but code is slightly simpler with your proposal.
I don't see any big issue or advantage in any of the both ways. So do as you wish.